docs/design/2021-08-23-history-hot-regions.md

```diff
@@ -0,0 +1,281 @@
+# Record history hotspot information in TiDB_HOT_REGIONS_HISTORY
+
+- Author(s): [@qidi1](https://github.com/qidi1), [@icepigzdb](http://github.com/icepigzdb)
+- Discussion PR: https://github.com/pingcap/tidb/pull/27487
+- Tracking Issue: https://github.com/pingcap/tidb/issues/25281
+
+## Table of Contents
+
+  * [Introduction](#introduction)
+  * [Motivation or Background](#motivation-or-background)
+  * [Detailed Design](#detailed-design)
+    + [Existing TIDB HOT REGIONS](#existing-tidb-hot-regions)
+    + [Table Header Design](#table-header-design)
+    + [PD](#pd)
+    + [TiDB](#tidb)
+  * [Test Design](#test-design)
+    + [Functional Tests](#functional-tests)
+    + [Scenario Tests](#scenario-tests)
+    + [Compatibility Tests](#compatibility-tests)
+    + [Benchmark Tests](#benchmark-tests)
+  * [Impacts & Risks](#impacts--risks)
+  * [Investigation & Alternatives](#investigation--alternatives)
+
+## Introduction
+
+Create a new table `TIDB_HOT_REGIONS_HISTORY` in the `INFORMATION_SCHEMAT` schema to retrieve history hotspot regions stored by PD periodically.  
+
+## Motivation or Background
+
+TiDB has a memory table `TIDB_HOT_REGIONS` that provides information about hotspot regions. But it only shows the current hotspot information. This leads to the fact that when DBAs want to query historical hotspot details, they have no way to find the corresponding hotspot information.
+
+According to the [documentation](https://docs.pingcap.com/tidb/stable/information-schema-tidb-hot-regions) for the current `TIDB_HOT_REGIONS` table,  it can only provides information about recent hotspot regions calculated by PD according to the heartbeat from tikv.  It is inconvenient to obtain hotspot regions of past time, and locate which store the region is. For ease of use, we can store extened hotspot region infomation in PD. The following list some common user cases:
```

Comment:
```suggestion
According to the [documentation](https://docs.pingcap.com/tidb/stable/information-schema-tidb-hot-regions) for the current `TIDB_HOT_REGIONS` table,  it can only provides information about recent hotspot regions calculated by PD according to the heartbeat from tikv.  It is inconvenient to obtain hotspot regions of past time, and locate which store the region is. For ease of use, we can store extended hotspot region information in PD. The following list some common use cases:
```

Reply:
change to proposal